src/main/java/org/assertj/core/internal/Arrays2D.java

```diff
@@ -0,0 +1,155 @@
+  }
+
+  @VisibleForTesting
+  public ComparisonStrategy getComparisonStrategy() {
+    return comparisonStrategy;
+  }
+
+  public static void assertIsArray(AssertionInfo info, Object array) {
+    if (!isArray(array)) throw Failures.instance().failure(info, shouldBeAnArray(array));
+  }
+
+  void assertNullOrEmpty(AssertionInfo info, Failures failures, Object array) {
+    if (array != null && !isArrayEmpty(array)) throw failures.failure(info, shouldBeNullOrEmpty(array));
+  }
+
+  void assertEmpty(AssertionInfo info, Failures failures, Object array) {
+    assertNotNull(info, array);
+    if (!isArrayEmpty(array)) throw failures.failure(info, shouldBeEmpty(array));
+  }
+
+  void assertHasSize(AssertionInfo info, Object array, int expectedSize) {
+    assertNotNull(info, array);
+    int sizeOfActual = sizeOf(array);
+    CommonValidations.checkSizes(array, sizeOfActual, expectedSize, info);
+  }
+
+  void assertHasSize(AssertionInfo info, Failures failures, Object array, int expectedSize, int firstDimensionIndex) {
+    assertNotNull(info, array);
+    int sizeOfActual = sizeOf(array);
+    checkSizes(array, failures, sizeOfActual, expectedSize, info, firstDimensionIndex);
+  }
+
+  private void checkSizes(Object actual, Failures failures, int sizeOfActual, int sizeOfOther, AssertionInfo info,
+    int firstDimensionIndex) {
+    if (sizeOfActual != sizeOfOther) throw failures.failure(info, shouldHaveSize(actual, sizeOfActual, sizeOfOther,
+      firstDimensionIndex));
+  }
+
+  public void assertHasSameSizeAs(AssertionInfo info, Object array, Object other) {
+    assertNotNull(info, array);
+    assertIsArray(info, array);
+    assertIsArray(info, other);
+    hasSameSizeAsCheck(info, array, other, sizeOf(array));
+  }
+
+  void assertContains(AssertionInfo info, Failures failures, Object array, Object value, Index index) {
+    assertNotNull(info, array);
+    assertNotEmpty(info, failures, array);
+    checkIndexValueIsValid(index, sizeOf(array) - 1);
+    Object actualElement = Array.get(array, index.value);
+    if (!areEqual(actualElement, value))
+      throw failures.failure(info, shouldContainAtIndex(array, value, index, Array.get(array, index.value),
+                                                        comparisonStrategy));
+  }
+
+  void assertNotEmpty(AssertionInfo info, Failures failures, Object array) {
+    assertNotNull(info, array);
+    if (isArrayEmpty(array)) throw failures.failure(info, shouldNotBeEmpty());
+  }
+
+  void assertNotEmpty(AssertionInfo info, Failures failures, Object array, ErrorMessageFactory failureMessage) {
+    assertNotNull(info, array);
+    if (isArrayEmpty(array)) throw failures.failure(info, failureMessage);
+  }
+
+  void assertDoesNotContain(AssertionInfo info, Failures failures, Object array, Object value, Index index) {
+    assertNotNull(info, array);
+    checkIndexValueIsValid(index, Integer.MAX_VALUE);
```

Comment:
should be `sizeOf(array) - 1` instead of `Integer.MAX_VALUE`, rationale being we need to be given a valid index

Reply:
Why? In class `src/main/java/org/assertj/core/internal/Arrays.java` this method also uses `Integer.MAX_VALUE`. In my opinion when array has 3 elements and we check if the array does not contain something at index 6, it should pass, because this is true. Of course we should use `sizeOf(array) - 1` in `assertContains`.